### PR TITLE
Fix UI: Improve add user into team flow #7789

### LINKED ISSUE #7789: UI: Improve add user into team flow
Currently, we are fetching all the users in add user model for adding users into the particular team,
- need to improve the current behaviour as it will cause an issue when there is 1000 user in the application

https://user-images.githubusercontent.com/71748675/192983012-b4eab430-97ac-4d29-9911-599c1a67bd68.mov

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes :

I worked on the Fixing UI: Improve add user into team flow #7789
Closes #7789 
#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/71748675/195321879-342c6381-b5f1-46f0-b4e4-9605b3c3b427.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/pages/teams/AddUsersModalV1.tsx [hunk 1]
```diff
@@ -140,70 +136,75 @@ const AddUsersModalV1 = ({
...
-      getUniqueUserList();
+  const onScroll = (e: React.UIEvent<HTMLElement, UIEvent>) => {
+    if (
+      e.currentTarget.scrollHeight - e.currentTarget.scrollTop ===
+      ContainerHeight
+    ) {
+      if (searchText) {
+        // make API call only when current page size is less then total count
+        PAGE_SIZE_MEDIUM * currentPage < totalESCount &&
+          searchUsers(searchText, currentPage);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/teams/AddUsersModalV1.tsx:156]: onScroll it shouldn't be searchUsers

File: openmetadata-ui/src/main/resources/ui/src/pages/teams/AddUsersModalV1.tsx [hunk 2]
```diff
@@ -140,70 +136,75 @@ const AddUsersModalV1 = ({
...
+      } else {
+        !isUndefined(userPaging.after) && fetchAllUsers(userPaging.after);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/teams/AddUsersModalV1.tsx:158]: We have this check twice
Other check is present in fetchAllUsers

**Review by chirag-madlani — COMMENTED** [review]
@ShaileshParmar11 Loading doesn't look good

File: openmetadata-ui/src/main/resources/ui/src/pages/teams/AddUsersModalV1.tsx [hunk 3]
```diff
@@ -11,81 +11,90 @@
...
+  const [currentPage, setCurrentPage] = useState(INITIAL_FROM);
+  const [totalESCount, setTotalESCount] = useState(0);
+
+  const searchUsers = (text: string, page: number) => {
+    searchData(text, page, PAGE_SIZE_MEDIUM, '', '', '', SearchIndex.USER)
+      .then((res: SearchResponse) => {
+        const data = formatUsersResponse(res.data.hits.hits);
+        setTotalESCount(res.data.hits.total.value);
+        setCurrentPage((pre) => pre + 1);
+        setUniqueUser((pre) => [...pre, ...data]);
```
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/teams/AddUsersModalV1.tsx:69]: Yes, We are fetching in a paginated way, so for keeping all the data we are keeping previous data and appending new data to it

File: openmetadata-ui/src/main/resources/ui/src/pages/teams/AddUsersModalV1.tsx (hunk 1 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/teams/AddUsersModalV1.tsx:156]: onScroll if there is a search term present, we need to fetch data from Search API, but if there is not search term then we are fetching from REST API

File: openmetadata-ui/src/main/resources/ui/src/pages/teams/AddUsersModalV1.tsx [hunk 4]
```diff
@@ -11,81 +11,90 @@
...
 type Props = {
+  isVisible: boolean;
   searchPlaceHolder?: string;
   header: string;
   list: Array<UserTeams>;
   onCancel: () => void;
   onSave: (data: Array<UserTeams>) => void;
 };
-
+const ContainerHeight = 250;
```
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/teams/AddUsersModalV1.tsx:47]: done

File: openmetadata-ui/src/main/resources/ui/src/pages/teams/AddUsersModalV1.tsx (hunk 2 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/teams/AddUsersModalV1.tsx:158]: Done

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**